homeassistant/components/sensibo/switch.py

```diff
@@ -0,0 +1,122 @@
+"""Switch platform for Sensibo integration."""
+from __future__ import annotations
+
+from collections.abc import Callable, Mapping
+from dataclasses import dataclass
+from typing import Any
+
+from pysensibo.model import SensiboDevice
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SensiboDataUpdateCoordinator
+from .entity import SensiboDeviceBaseEntity
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass
+class DeviceBaseEntityDescriptionMixin:
+    """Mixin for required Sensibo base description keys."""
+
+    value_fn: Callable[[SensiboDevice], bool | None]
+    extra_fn: Callable[[SensiboDevice], dict[str, str | bool | None]]
+
+
+@dataclass
+class SensiboDeviceSwitchEntityDescription(
+    SwitchEntityDescription, DeviceBaseEntityDescriptionMixin
+):
+    """Describes Sensibo Switch entity."""
+
+
+DEVICE_SWITCH_TYPES: tuple[SensiboDeviceSwitchEntityDescription, ...] = (
+    SensiboDeviceSwitchEntityDescription(
+        key="timer_on_switch",
+        device_class=SwitchDeviceClass.SWITCH,
+        name="Timer",
+        icon="mdi:timer",
+        value_fn=lambda data: data.timer_on,
+        extra_fn=lambda data: {"id": data.timer_id, "turn_on": data.timer_state_on},
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sensibo binary sensor platform."""
+
+    coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[SensiboDeviceSwitch] = []
+
+    entities.extend(
+        SensiboDeviceSwitch(coordinator, device_id, description)
+        for description in DEVICE_SWITCH_TYPES
+        for device_id, device_data in coordinator.data.parsed.items()
+        if device_data.model != "pure"
+    )
+
+    async_add_entities(entities)
```

Comment:
```suggestion
    async_add_entities(
        SensiboDeviceSwitch(coordinator, device_id, description)
        for description in DEVICE_SWITCH_TYPES
        for device_id, device_data in coordinator.data.parsed.items()
        if device_data.model != "pure"
    )

```

Reply:
But if you are going to add more probably best to keep it as-is

Reply:
Should move pure boost here as well as you wrote in the comment below.